### ISSUE TITLE
Could you add an option to use TCP

### ISSUE THREAD

**Issue by emdee-is** [metebalci/digsec#9]
Could you add an option to use TCP? For those behind TCP only firewalls.

It's often on 53/tcp but if you could add a port option, so much the better.

**Comment by metebalci** [comment]
Not very difficult to add but I think using a different port for DNS is a bit strange. I should add tcp.

**Comment by emdee-is** [comment]
The reason for the "strange" port is a unixism: 53 is < 1024 and only root can do that. So if you have an alternate DNS, it can't run as it's own user if it wants to bind to 53. So binding another port is common when you don't want the thing running as root.

**Comment by metebalci** [comment]
If you are running DNS at non-53 port, you have a problem with firewalls anyway. If it is a lab environment, you can redirect the port.

**Comment by emdee-is** [comment]
2,500,000 people know that you are wrong; they're running behind a TCP only proxy that runs a UDP DNS server by convention on 9053, but by requirement on a port > 1023. And they *rely* on it.

Besides dig supports it...

**Comment by metebalci** [comment]
I do not know what environment you are working, I have not seen any production environment where DNS is run at non standard ports. I dont think there is a DHCP option for DNS port, and it is not possible to advertise DNS port with RA in IPv6, if you know otherwise please correct me.

Thanks for comparing it to dig anyway but dig supports it because DNS may require it, without EDNS0 the message size is limited  particularly for DNSSEC and it can be truncated so the client should switch to TCP. I am working on it because of that, not because it might be needed in an environment.

**Comment by emdee-is** [comment]
You stand corrected: https://metrics.torproject.org/userstats-relay-country.html

They are all using DNS as UDP or TCP on 9053.

**Comment by metebalci** [comment]
It is not supposed to be used behind Tor.

**Comment by emdee-is** [comment]
Says who? It works fine behind Tor, or any proxy that offers an alternate DNS.

If it is not supposed to work behind proxies then you are saying it is not supposed to be used in any corporate environment: they're all behind proxies.

**Comment by metebalci** [comment]
I say, as the owner of this project. What I mean is it is not supported, I dont have a way of checking this, neither I have interest or time at the moment. So it should be used as it is intended for lets say normal/traditional internet environment.

At corporate environment, usually http/layer-7 is proxied, which is different, they use standard DNS. Even if layer 3-4 is proxied, and if it is SOCKS proxy, still standard DNS is used. That is why I am saying, so many things depend on standard DNS ports, it is not that easy to say lets change it. It is not like using http at 80 or 8080.

**Comment by metebalci** [comment]
TCP support added with bd2580aca6037d93ac48dea8942991e0730fd650

I will add port as well, then I will close the issue

**Comment by metebalci** [comment]
non-53 port support added with 972e0413f0aff40dd484c9eab633a75244a76706, but I have no way to verify it at the moment

**Comment by metebalci** [comment]
I will release a new one on pip sometime, not sure but maybe today, if you want you can wait that as well

**Comment by emdee-is** [comment]
Why not leave it for monday and leave me the weekend to test out the current changes. It may take be a bit to do the TCP testing.

The test.sh is completely useless - you misunderstood what I meant about reordering and adding ```set -e``` to ```validate_second_level_domain.sh```

Would you consider adding a ```validate_second_level_domain.sh``` - yours or "mine"?

I greatly appreciate your incredible response to my issues.

**Comment by metebalci** [comment]
Not sure what you are asking about validate script. I have added a new validate.py

Just to make sure we are on the same page. I am not implementing what you are asking. Without knowing how DNSSEC works, reordering the validation steps mean nothing. It should be according to how RFC says it. (on the other hand, it is a chain with trust anchor, it should not matter how you authenticate the responses for a valid chain, only for invalid chain the position of error I guess changes)

**Comment by emdee-is** [comment]
Writing a python program to use``` os.system``` to call a python program is considered in very poor taste :-( 

At least you could fix scripts.main() to take ```sys.argv[1:]``` when called from ```__name__ == '__main__'``` as an argument and then you could
```
from digsec.scripts import main
main('your command line'.split())
```
or better still
```
import shlex
main(shlex.split('your command line'))
```
It's would be more efficient and would preserve the class of errors

**Comment by emdee-is** [comment]
Non-53 port tested and works, with some surprising results.

**Comment by metebalci** [comment]
This is not a python application code, it is just a way of writing a script since it will not be as straight-forward as like this if I try it in bash.
